tools/travis_before_install.sh

```diff
@@ -28,33 +28,49 @@ retry () {
 }
 
 
-# on Python 2.7, use the system versions of numpy, scipy, and matplotlib
-# and the minimum version of cython and networkx
-if [[ $TRAVIS_PYTHON_VERSION == 2.7* ]]; then
+# on Python 2.7, use the system versions of matplotlib, numpy and scipy
+# which is also our minimum requirements
+# this will change if travis updates to ubuntu LTS 14.04
+# http://docs.travis-ci.com/user/ci-environment/#CI-environment-OS
+if [ $TRAVIS_PYTHON_VERSION == 2.7 ]; then
+    sudo apt-get install python-scipy python-matplotlib python-numpy python-imaging
+    # above should be equal to minimum requirements
+    # remove from requirements and set the rest to minimum version
+    sed -i '/^matplotlib.*$/d' requirements.txt
+    sed -i '/^scipy.*$/d' requirements.txt
+    sed -i '/^numpy.*$/d' requirements.txt
+    sed -i '/^pillow.*$/d' requirements.txt
+    sed -i 's/>=/==/g' requirements.txt
+    # remove pillow from wheel install
+    export WHEELBINARIES=${WHEELBINARIES/pillow/}
+    # use system python
     virtualenv --system-site-packages ~/venv
-    sudo apt-get install python-scipy python-matplotlib python-imaging
-    sed -i 's/cython>=/cython==/g' requirements.txt
-    sed -i 's/networkx>=/networkx==/g' requirements.txt
-    sed -i '/pillow/d' requirements.txt
+    # install system setuptools, avoiding compiled pkg_resources from travis env
+    sudo apt-get install python-setuptools
 else
-    virtualenv -p python --system-site-packages ~/venv
+    virtualenv -p $(which python) ~/venv
```

Comment:
I think the `$(which python)` is what is breaking your build, try reverting this line.